lib/propolis/src/hw/chipset/i440fx.rs

```diff
@@ -366,17 +171,100 @@ impl Piix4HostBridge {
             ..Default::default()
         })
         .finish();
-        Arc::new(Self { pci_state })
+
+        let pin_power = opts.power_pin.unwrap_or_else(|| Arc::new(NoOpPin {}));
+        let pin_reset = opts.reset_pin.unwrap_or_else(|| Arc::new(NoOpPin {}));
+
+        let pci_cfg = PioCfgDecoder::new();
+        let pcie_cfg = opts.enable_pcie.then(|| {
+            PcieCfgDecoder::new(pci::bits::PCIE_MAX_BUSES_PER_ECAM_REGION)
+        });
+
+        Arc::new(Self {
+            pci_state,
+
+            pci_topology,
+            pci_cfg,
+            pcie_cfg,
+
+            pin_power,
+            pin_reset,
+        })
+    }
+
+    pub fn attach(self: &Arc<Self>, machine: &Machine) {
```

Comment:
Definitely a "Doc it hurts when I do this.  Then don't do that" situation for now.  I'd like to see a better abstraction around MMIO/PIO attachments at some point, which would make it easier to avoid such mistakes.